addons/account/models/account_move.py

```diff
@@ -1708,11 +1708,26 @@ def _check_balanced(self):
         if not moves:
             return
 
+        postcommit_check_balanced = self.env.context.get('postcommit_check_balanced')
+        if postcommit_check_balanced == []:
+            # Add postcommit hook only once
+            @self.env.cr.postcommit.add
```

Comment:
You commit something wrong then say that it is wrong  to the user but keep the wrong values?

Reply:
I don't quite understand you question.
`postcommit` is executed after successful commit only, right?

Reply:
Exactly?!
That's the very issue, the commit will be successful, but we don't want it to be, because we need to execute this constraint **before** the commit.

Reply:
Indeed. It it should be executed before the commit

Reply:
fixed